src/conventions/h-group/interpret-discard.js

```diff
@@ -97,15 +98,26 @@ export function interpret_discard(game, action, card) {
 	// Discarding a useful card
 	// Note: we aren't including chop moved and finessed cards here since those can be asymmetric.
 	// Discarding with a finesse will trigger the waiting connection to resolve.
-	if (card.clued && rank > state.play_stacks[suitIndex] && rank <= state.max_ranks[suitIndex]) {
-		logger.warn('discarded useful card!');
-		common.restore_elim(card);
-
-		// Card was bombed
-		if (failed)
-			undo_hypo_stacks(game, identity);
-		else
-			interpret_sarcastic(game, action);
+	if (rank > state.play_stacks[suitIndex] && rank <= state.max_ranks[suitIndex]) {
+		if (card.clued) {
+			logger.warn('discarded useful clued card!');
+			common.restore_elim(card);
+
+			// Card was bombed
+			if (failed)
+				undo_hypo_stacks(game, identity);
+			else
+				interpret_sarcastic(game, action);
```

Comment:
We are not in dda if there was a successful *Sarcastic Discard*, so we probably need to check the return value from this to decide.

Reply:
Good catch, done!